.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,139 @@
+name: Nightly
+
+on:
+  workflow_dispatch:
+
+jobs:
+  retrieve-info:
+    runs-on: ubuntu-22.04
+    outputs:
+      artifacts-link: ${{ steps.artifacts.outputs.link }}
+      version: ${{ steps.get-version.outputs.version }}
+    steps:
+      - name: Retrieve artifacts url
+        uses: scality/action-artifacts@v3
+        id: artifacts
+        with:
+          method: get
+          workflow-name: Pre-merge
+          url: https://artifacts.scality.net
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+      - name: Retrieve product.txt from artifacts
+        run: >
+          curl --fail -LO -u ${{ secrets.ARTIFACTS_USER }}:${{ secrets.ARTIFACTS_PASSWORD }}
+          ${{ steps.artifacts.outputs.link }}/product.txt
+      - name: Retrieve current version
+        id: get-version
+        run: cat product.txt >> $GITHUB_OUTPUT
+
+  compute-lifecycle-matrix:
+    runs-on: ubuntu-22.04
+    needs:
+      - retrieve-info
+    env:
+      VERSION: ${{ needs.retrieve-info.outputs.version }}
+    outputs:
+      # Every element of the matrix will have
+      # - `type` which is the type of lifecycle (major, minor, patch)
+      # - `version` the promoted version
+      lifecycle-matrix: ${{ steps.get-matrix.outputs.matrix }}
+      artifacts-link: ${{ needs.retrieve-info.outputs.artifacts-link }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Fetch tags
+        run: git fetch --tags
+      - name: Retrieve current version
+        run: |
+          echo "major=$(echo "$VERSION" | cut -d'.' -f1)" >> $GITHUB_ENV
+          echo "minor=$(echo "$VERSION" | cut -d'.' -f2)" >> $GITHUB_ENV
+          echo "patch=$(echo "$VERSION" | cut -d'.' -f3)" >> $GITHUB_ENV
```

Comment:
> Why do we need to go this convoluted route (fetching product.txt from artifacts, getting its VERSION, and parse it into components) while we have the `VERSION` file at hand?

We could, I'm using `product.txt` so that we can basically use any artifacts to run it, but here indeed we will likely always be linked to the branch where we are running

> Also, if I'm not mistaken, the parsing will always have `-dev` in the resulting `patch` (since we are running the nightly on development branches).

Right, I will change it (it works when patch > 1 since `$ "1-dev" - 1))` give `0` but when it will be `0-dev` it will not work properly)

I will switch to use the repository version file